sdk/daml-lf/transaction-test-lib/src/main/scala/com/digitalasset/daml/lf/transaction/test/TreeTransactionBuilder.scala

```diff
@@ -52,7 +52,7 @@ object TreeTransactionBuilder extends TreeTransactionBuilder {
 
   private final case class Tx(nodes: Map[NodeId, Node], roots: Seq[NodeId]) {
     private def version =
-      nodes.values.flatMap(_.optVersion).headOption.getOrElse(TransactionVersion.VDev)
+      nodes.values.flatMap(_.optVersion).maxOption.getOrElse(TransactionVersion.maxVersion)
```

Comment:
Doesn't this change the semantics of this function? Or are all nodes supposed to have the same version anyway?

Reply:
this as a bug. Transaction version should be higher or equal to all the nodes.